From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: okashaluai <luaiokasha@gmail.com>
Date: Wed, 30 Sep 2026 15:39:20 +0200
Subject: [PATCH] 9696: docs: Clarify the ^ concatenator behavior in Sleigh
 display sections

docs: Clarify the ^ concatenator behavior in Sleigh
 display sections

applied the documentation changes suggestions to the sleigh.xml - responsible for generating the html files
---
 Ghidra/Features/Decompiler/src/main/doc/sleigh.xml | 6 +++---
 GhidraDocs/languages/html/sleigh_constructors.html | 6 +++---
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/Ghidra/Features/Decompiler/src/main/doc/sleigh.xml b/Ghidra/Features/Decompiler/src/main/doc/sleigh.xml
index 6cf88e6a72..79ecd28edd 100644
--- a/Ghidra/Features/Decompiler/src/main/doc/sleigh.xml
+++ b/Ghidra/Features/Decompiler/src/main/doc/sleigh.xml
@@ -1552,8 +1552,8 @@ no such requirement.
 <sect3 id="sleigh_caret">
 <title>The '^' character</title>
 <para>
-The ‘^’ character in the display section is used to separate
-identifiers from other characters where there shouldn’t be white space
+The ‘^’ character in the display section is used to concatenate
+identifiers and literal characters where there shouldn’t be white space
 in the disassembly display. This can be used in any manner but is
 usually used to attach display characters from a local symbol to the
 literal characters of the mnemonic.
@@ -1567,7 +1567,7 @@ literal characters of the mnemonic.
 In the above example, “bra” is treated as literal characters in the
 resulting display string followed immediately, with no intervening
 spaces, by the display string of the local
-symbol <emphasis>cc</emphasis>. Thus the whole constructor actually
+symbol <emphasis>cc</emphasis> (resulting in an output like braeq or brane). Thus, even though it is attached to the literal bra, the whole constructor actually
 has three operands, denoted by the three
 identifiers <emphasis>cc</emphasis>, <emphasis>op1</emphasis>,
 and <emphasis>op2</emphasis>.
diff --git a/GhidraDocs/languages/html/sleigh_constructors.html b/GhidraDocs/languages/html/sleigh_constructors.html
index 2ab46368c1..bb49218b26 100644
--- a/GhidraDocs/languages/html/sleigh_constructors.html
+++ b/GhidraDocs/languages/html/sleigh_constructors.html
@@ -232,8 +232,8 @@ no such requirement.
 <div class="titlepage"><div><div><h4 class="title">
 <a name="sleigh_caret"></a>7.3.2. The '^' character</h4></div></div></div>
 <p>
-The ‘^’ character in the display section is used to separate
-identifiers from other characters where there shouldn’t be white space
+The ‘^’ character in the display section is used to concatenate
+identifiers and literal characters where there shouldn’t be white space
 in the disassembly display. This can be used in any manner but is
 usually used to attach display characters from a local symbol to the
 literal characters of the mnemonic.
@@ -247,7 +247,7 @@ literal characters of the mnemonic.
 In the above example, “bra” is treated as literal characters in the
 resulting display string followed immediately, with no intervening
 spaces, by the display string of the local
-symbol <span class="emphasis"><em>cc</em></span>. Thus the whole constructor actually
+symbol <span class="emphasis"><em>cc</em></span> (resulting in an output like braeq or brane). Thus, even though it is attached to the literal bra, the whole constructor actually
 has three operands, denoted by the three
 identifiers <span class="emphasis"><em>cc</em></span>, <span class="emphasis"><em>op1</em></span>,
 and <span class="emphasis"><em>op2</em></span>.
-- 
2.45.1

